File: Day29/cpp/print_N_Times.cpp
#include<bits/stdc++.h>
using namespace std;

void printString(int i ,int n,vector<string> &str){
	if(n<i) return;
	else{
		str.push_back("Coding Ninjas");
		printString(i+1,n,str);
	}
}
vector<string> printNTimes(int n) {
	vector<string>str;
	printString(1,n,str);
	return str;
}

int main()
{
    int n;
    cin>>n;
    vector<string> str = printNTimes(n);
    for(int i=0;i<str.size();i++){
        cout<<str[i]<<endl;
    }
    
    return 0;
}

File: Day29/java/print_N_Times.java
package Day29.java;

import java.util.ArrayList;
import java.util.List;

public class print_N_Times {
    static void printvalues(int i,int x,List<String> str){
        if(x<i) return;
        else{
            str.add("Coding Ninjas");
            printvalues(i+1, x, str); 
        }
    }
    public static List<String> printNtimes(int n){
        List<String> str = new ArrayList<String>();
        printvalues(1, n, str);
        return str;
    }
    public static void main(String[] args) {
        List<String> str = printNtimes(5);
        for(String s:str){
            System.out.println(s);
        }
    }
}
